lib/grape/middleware/base.rb

```diff
@@ -27,7 +29,21 @@ def call!(env)
         ensure
           after_response = after
         end
-        after_response || @app_response
+
+        response = after_response || @app_response
+        reflect_headers response
+        response
+      end
+
+      def reflect_headers(response)
+        return unless header.is_a?(Hash)
```

Comment:
What else can `header` be? I think a check for `nil` is more appropriate, so `unless header`. Also alias `header` to `headers`, when we check on a collection of headers we should be doing `headers`.